x/protocolpool/keeper/keeper.go

```diff
@@ -90,6 +101,151 @@ func (k Keeper) GetCommunityPool(ctx context.Context) (sdk.Coins, error) {
 	return k.bankKeeper.GetAllBalances(ctx, moduleAccount.GetAddress()), nil
 }
 
+func (k Keeper) withdrawContinuousFund(ctx context.Context, recipient sdk.AccAddress) (sdk.Coin, error) {
+	cf, err := k.ContinuousFund.Get(ctx, recipient)
+	if err != nil {
+		if errors.Is(err, collections.ErrNotFound) {
+			return sdk.Coin{}, fmt.Errorf("no continuous fund found for recipient: %s", recipient.String())
+		}
+		return sdk.Coin{}, err
+	}
+
+	toDistributeAmount, err := k.ToDistribute.Get(ctx)
+	if err != nil {
+		return sdk.Coin{}, err
+	}
+	if toDistributeAmount.Equal(math.ZeroInt()) {
+		return sdk.Coin{}, nil
+	}
+
+	denom, err := k.iterateAndUpdateFundsDistribution(ctx, toDistributeAmount)
+	if err != nil {
+		return sdk.Coin{}, fmt.Errorf("error while iterating all the continuous funds: %w", err)
+	}
+	// get allocated continuous fund
+	fundsAllocated, err := k.getDistributedFunds(ctx, recipient)
```

Comment:
Is this helper still needed now?

Reply:
nope, cleanup still needs to be done. Once after designing phase is practically approved, I'll go through the whole code and docs for a better cleanup.
I'll remove this for now